controllers/heatcfnapi_controller.go

```diff
@@ -810,6 +797,38 @@ func (r *HeatCfnAPIReconciler) reconcileUpgrade(ctx context.Context, instance *h
 	return ctrl.Result{}, nil
 }
 
+// getSecret - get the specified secret, and add its hash to envVars
+func (r *HeatCfnAPIReconciler) getSecret(
```

Comment:
note this approach with getSecret is copied from designate which has these functions in each controller as well.  we can try to improve this but it would add more dev time